infrastructure/lambda-generate-stitch-record.tf

```diff
@@ -0,0 +1,131 @@
+module "generate-stitch-record-alarm" {
+  source               = "./modules/lambda_alarms"
+  lambda_function_name = module.generate-stitch-record-lambda.function_name
+  lambda_timeout       = module.generate-stitch-record-lambda.timeout
+  lambda_name          = "generate_stitch_record_handler"
+  namespace            = "AWS/Lambda"
+  alarm_actions        = [module.generate-stitch-record-alarm-topic.arn]
+  ok_actions           = [module.generate-stitch-record-alarm-topic.arn]
+  depends_on           = [module.generate-stitch-record-lambda, module.generate-stitch-record-alarm-topic]
+}
+
+module "generate-stitch-record-alarm-topic" {
+  source                = "./modules/sns"
+  sns_encryption_key_id = module.sns_encryption_key.id
+  current_account_id    = data.aws_caller_identity.current.account_id
+  topic_name            = "generate-stitch-record-topic"
+  topic_protocol        = "lambda"
+  topic_endpoint        = module.generate-stitch-record-lambda.lambda_arn
+  delivery_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "cloudwatch.amazonaws.com"
+        },
+        "Action" : [
+          "SNS:Publish",
+        ],
+        "Condition" : {
+          "ArnLike" : {
+            "aws:SourceArn" : "arn:aws:cloudwatch:eu-west-2:${data.aws_caller_identity.current.account_id}:alarm:*"
+          }
+        }
+        "Resource" : "*"
+      }
+    ]
+  })
+
+  depends_on = [module.generate-stitch-record-lambda, module.sns_encryption_key]
+}
+
+module "generate-stitch-record-lambda" {
+  source                   = "./modules/lambda"
+  name                     = "GenerateStitchRecord"
```

Comment:
Can we call this `GenerateLloydGeorgeStitch` or `LloydGeorgeStitchJob`? So there is some consistency between this and the `LloydGeorgeStitch` lambda?